### PR TITLE
[1283] Add `application_status` to `course` table

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

This is the first card in the removing vacancies epic and sets out the skeleton for the work to come. 

### Changes proposed in this pull request

- Migration to add `application_status` to the course table as an integer
- Enum helper methods to allow for `Course.open?` and `Course.closed?` etc.

### Guidance to review

- Are the default values sensible?
- Should we have a `null: false` on the column, given that we set a default value of `0`?

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

**Comment by avinhurry** [comment]
[Add new open or closed column for courses](https://trello.com/c/EDiyG99J/1283-add-new-open-or-closed-column-for-courses)

File: db/migrate/20230519144435_add_application_status_to_course.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddApplicationStatusToCourse < ActiveRecord::Migration[7.0]
+  def change
+    add_column :course, :application_status, :integer, default: 0, null: false
```
Comment (darokel) [inline db/migrate/20230519144435_add_application_status_to_course.rb:5]: let's an index here, we'll definitely be searching against this (from Find)

Comment (avinhurry) [reply db/migrate/20230519144435_add_application_status_to_course.rb]: Yup, done: https://github.com/DFE-Digital/publish-teacher-training/pull/3578/commits/8f1e90b28533a256f853ab2c6886368838abed90

File: db/migrate/20230519144435_add_application_status_to_course.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class AddApplicationStatusToCourse < ActiveRecord::Migration[7.0]
+  def change
+    add_column :courses, :application_status, :integer, default: 0, null: false
+
+    add_index :courses, :application_status
```
Comment (defong) [inline db/migrate/20230519144435_add_application_status_to_course.rb:7]:
```suggestion
    add_column :course, :application_status, :integer, default: 0, null: false

    add_index :course, :application_status
```

**Review by defong — CHANGES_REQUESTED** [review]
Does not work

**Review by defong — APPROVED** [review]

**Review by darokel — APPROVED** [review]

**Pull request merged**